tests/test-user.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+require("dotenv").load();
```

Comment:
Do we use any ENV vars here that we need loaded via `dotenv`? Or does ../db/models require something?

Reply:
No, we don't need `dotenv` here. In fact, I'm starting to dislike `dotenv` and `AppConstants` in general. Because other libraries (e.g., Sequelize) are written to use actual `process.env` variables - not use the `dotenv` magic.

I may file an issue to remove `dotenv`, and perhaps refactor `AppConstants` too.

Reply:
I'm torn. I like [**node-convict**](https://github.com/mozilla/node-convict) for managing configs and flags, but it can get a bit verbose. But I'd be curious to see your proposal for replacing .env files and keeping the 20+ env var configs sane.

Another big plus for **convict** is that you could have separate config files for "development", "production", and "testing", if we wanted to store different values per environment (although it may not be wise checking those in to source control). But still having all my local API keys and flags set in a local "./configs/development.json" would be useful).

Reply:
From what I see, I also like `node-convict` more than `dotenv`. It seems more sane to put 20+ configs in the single `config.js` module with default values than to declare and define them across `.env-dist`, `.env`, and `app-constants.js`.

Reply:
FWIW, I have only ever used `.env` files along with [`autoenv`](https://github.com/kennethreitz/autoenv) for a local/development convenience. I've never used `.env` files for staging or production app servers. So, my local `.env` looks like:
```
export DATABASE_URL=<local-dev-value>
export DEBUG_DUMMY_SMTP=1
...
```
And `autoenv` simply runs those `export` statements when I `cd` into the working directory.

The app code itself *always* pulls from environment variables or default values - never depends on `.env` files.